zephyr/include/rtos/wait.h

```diff
@@ -12,32 +12,31 @@
 #include <stdbool.h>
 #include <zephyr/kernel.h>
 
-/* TODO: use equivalent Zephyr */
 static inline void idelay(int n)
 {
-	while (n--)
-		asm volatile("nop");
+	k_busy_wait((uint32_t)n);
 }
 
 /* DSP default delay in cycles - all platforms use this today */
 #define PLATFORM_DEFAULT_DELAY	12
 
 static inline void wait_delay(uint64_t number_of_clks)
 {
-	uint64_t timeout = sof_cycle_get_64() + number_of_clks;
+	uint64_t timeout;
 
-	while (sof_cycle_get_64() < timeout)
-		idelay(PLATFORM_DEFAULT_DELAY);
+	timeout = number_of_clks * NSEC_PER_SEC / CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC;
```

Comment:
CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC is not the DSP clock, so I think the resulting effective delay is different after this PR.